src/audio/mux/mux.c

```diff
@@ -200,7 +204,217 @@ static struct comp_dev *mux_new(const struct comp_driver *drv,
 	dev->state = COMP_STATE_READY;
 	return dev;
 }
+#else
+static int build_config(struct comp_data *cd, struct comp_dev *dev)
+{
+	dev->ipc_config.type = SOF_COMP_MUX;
+	cd->config.frame_format_deprecated = cd->md.base_cfg.audio_fmt.depth;
+	cd->config.num_streams = 2;
+	int mask = 1;
+
+	/* clear masks */
+	for (uint16_t i = 0; i < cd->config.num_streams; i++)
+		for (uint16_t j = 0 ; j < PLATFORM_MAX_CHANNELS; j++)
+			cd->config.streams[i].mask[j] = 0;
+	/*
+	 * Setting masks for streams. Channels for IPC4 are mapped
+	 * according to pin and channel order. First the first input stream, then the
+	 * reference. For example, for base config 2-channel and reference 2-channel, masks
+	 * should look like: mask[] = { 0b00000001, 0b00000010 } for the first
+	 * stream and mask[] = { 0b00000100, 0b00001000 } for the second
+	 * (reference). This means that the first channel of first stream
+	 * will be copied to the first output channel, second channel of the first stream
+	 * will be copied to the second output channel, first channel of second stream
+	 * will be copied to the third output channel, the second channel of second stream
+	 * will be copied to the fourth output channel.
+	 */
+	for (uint16_t i = 0; i < cd->md.base_cfg.audio_fmt.channels_count; i++) {
+		cd->config.streams[0].mask[i] = mask;
+		mask = mask * 2;
+	}
+
+	for (uint16_t i = 0; i < cd->md.reference_format.channels_count; i++) {
+		cd->config.streams[1].mask[i] = mask;
+		mask = mask * 2;
+
+	}
+
+	/* verification that masks are correct */
+	for (uint16_t i = 0; i < cd->config.num_streams; i++) {
+		for (uint16_t j = 0 ; j < PLATFORM_MAX_CHANNELS; j++) {
+			if (popcount(cd->config.streams[i].mask[j]) > 1) {
+				comp_cl_err(&comp_mux, "build_config(): mux component is not able to mix channels");
```

Comment:
I think this error message is confusing, maybe `build_config(): invalid mask for stream %d, channel:%d`?